card-function

sammurraytuesta opened the pull request:
New changes:
- Added card functionality using a form submission to determine the text to present on card. Text is pulled from the form using JavaScript which prevents the default form submission, grabs the text and inputs it into a variable that is later concatenated into the innerHTML of a created div called "note."
- Implemented basic UI to support the visual and technical display of each card and its form submission
- Renamed files to popup.<filetype> to add uniformity (except for manifest.json, and README.md, as they are specific files with specific uses that involve their name)
- Added "default_icon" values to "browser_action" in manifest.json in accordance with chrome extension common practices

sammurraytuesta commented on css/popup.css:
```diff
@@ -0,0 +1,155 @@
+.edits-container .bi:hover {
+    color: var(--highlight);
+    cursor: pointer;
+}
```
Woo!

sammurraytuesta commented on popup.js:
```diff
@@ -0,0 +1,52 @@
+    var submitNote = document.getElementById("submit");
+    var text = "";
+
+    /* Creates a form that allows for the user to input text to later be used in a note. */
```
Thank you for the link! I adjusted accordingly! If things still don't look correct let me know!

sammurraytuesta commented on popup.js:
```diff
@@ -0,0 +1,52 @@
+        var newNote = document.createElement('div');
+        newNote.setAttribute("class", "note");
+
+        /* Adds in the inner html that is basically the note. CSS takes these elements into account. Text variable holds the text from the above function that creates an instance of a note */
```
Thank you! I installed the Beautify extension but it didn't format the commenting all that much! I took a look at the Google JavaScript Style Guide for adjusting the comments! The extension did add a new line at the end of the document though!

sammurraytuesta commented on popup.js:
```diff
@@ -0,0 +1,52 @@
+window.onload = function () {
+    /* GLOBAL VARIABLES: form, add, submit, and text grabbed from submission of the form. */
```
Created a class to hold the global variables within a constructor! Variables are accessed directly from the class, would you rather I use accessors?

sammurraytuesta commented on popup.js:
```diff
@@ -0,0 +1,52 @@
+        newNote.setAttribute("class", "note");
+
+        /* Adds in the inner html that is basically the note. CSS takes these elements into account. Text variable holds the text from the above function that creates an instance of a note */
+        newNote.innerHTML = "<div class='text'>" + text + "</div><div class='edits-container'><i id='edit' class='bi bi-pencil'></i><i id='delete' class='bi bi-trash'></i></div>";
```
Thank you for the suggestion! Now when the user inputs something like <h1>Hello World!</h1> it will turn out exactly as written in the note rather than actually populate that element in the card! It was very cool implementing a template!

rjkroege commented on popup.js:
```diff
@@ -1,52 +1,70 @@
 
-        /* Adds in the inner html that is basically the note. CSS takes these elements into account. Text variable holds the text from the above function that creates an instance of a note */
-        newNote.innerHTML = "<div class='text'>" + text + "</div><div class='edits-container'><i id='edit' class='bi bi-pencil'></i><i id='delete' class='bi bi-trash'></i></div>";
+        //Appends child (clone of note template) to the top.
```
Should have a space after the `//`

sammurraytuesta replied:
I just added the space to this comment and all others of the like that require a space after the comment indication! Thank you!

rjkroege commented on popup.js:
```diff
@@ -1,52 +1,70 @@
-        if (plusNote.classList.contains("bi-x-circle")) {
-            plusNote.classList.remove("bi-x-circle");
-            plusNote.classList.add("bi-plus-circle");
+        //Takes care of consistency in the plus vs x button for creating/cancelling a note.
```
Ditto.

sammurraytuesta replied:
Thanks!

rjkroege reviewed (commented):
Code changes seem good. Be sure to clean up the formatting.

rjkroege commented:
You need to rebase this.

rjkroege commented:
And (ideally) you will fix some formatting issues before landing but looks good. I'm glad that the template idea worked for you. 😀

sammurraytuesta commented:
I was able to update this PR by merging the changes from the main branch and combining them with the new changes in the card-function branch. Because I renamed some files for consistency, I moved the content from the previously named files to the current files to avoid duplicates! Everything should be up to date with Santiago's changes now! I also updated the formatting so hopefully it all looks good but if not let me know and I'd be happy to make further changes! Thank you!

rjkroege commented on popup.html:
```diff
@@ -0,0 +1,101 @@
+<!DOCTYPE html>
```
Santiago's changes shouldn't be in your changes. I think that you have made a mistake.

sammurraytuesta replied:
I merged the updated changes from the main branch into mine, is that what you mean by rebase? Or do you mean call git rebase and add my changes to that? Because when I tried to use git rebase all of my changes were lost. Is there a different approach that you would recommend?

rjkroege commented:
I will give you better comments later about this. Sorry for the delay.

sammurraytuesta commented:
All good no worries!

rjkroege commented:
I had misread part of this. LGTM.
